src/builder.rs

```diff
@@ -0,0 +1,141 @@
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+
+use anyhow::Result;
+use hyper::{Body, Request, Response};
+
+use crate::app_context::AppContext;
+use crate::async_graphql_hyper::{GraphQLBatchRequest, GraphQLRequest};
+use crate::blueprint::{validate_operations, Blueprint, OperationQuery};
+use crate::config::{Config, ConfigModule, ConfigReader, Source};
+use crate::fmt::Fmt;
+use crate::http::handle_request;
+use crate::print_schema;
+use crate::runtime::TargetRuntime;
+use crate::valid::Validator;
+
+/// Struct to build TailcallExecutor
+#[derive(Clone)]
+pub struct TailcallBuilder {
+    /// Holds a list of file paths
+    files: Vec<String>,
+    /// Holds a list of schema information
+    schemas: Vec<SchemaHolder>,
+}
+
+/// Struct to hold schema information while building executor
+#[derive(Clone)]
+struct SchemaHolder {
+    /// Holds a type of schema
+    source: Source,
+    /// Holds content of schema
+    schema: String,
+    /// Holds a path to parent dir for content in @link
+    parent_dir: Option<PathBuf>,
+}
+
+#[derive(Clone)]
+pub struct TailcallExecutor {
+    pub config_module: ConfigModule,
+    pub app_ctx: Arc<AppContext>,
```

Comment:
Ideally we should not keep fields in executors public.

Reply:
dropped ConfigModule and I reduced public use of AppContext, now it's just used in `handle_request` for showcase and in tests.